exo/api/chatgpt_api.py

```diff
@@ -198,6 +200,9 @@ async def middleware(request):
   async def handle_root(self, request):
     return web.FileResponse(self.static_dir/"index.html")
 
+  async def handle_model_support(self, request):
+    return web.json_response({"model pool": { m: pretty_name.get(m, m) for m in [k for k,v in model_cards.items() if all(map(lambda e: e in v["repo"], list(dict.fromkeys([inference_engine_classes.get(i,None) for i in self.node.topology_inference_engines_pool for i in i if i is not None] + [self.inference_engine_classname]))))]}})
```

Comment:
gnarly, but would be far more readable if you break it up with line breaks:

```
  async def handle_model_support(self, request):
    return web.json_response({
      "model pool": {
        model_name: pretty_name.get(model_name, model_name) 
        for model_name in [
          model_id for model_id, model_info in model_cards.items() 
          if all(map(
            lambda engine: engine in model_info["repo"],
            list(dict.fromkeys([
              inference_engine_classes.get(engine_name, None) 
              for engine_list in self.node.topology_inference_engines_pool 
              for engine_name in engine_list 
              if engine_name is not None
            ] + [self.inference_engine_classname]))
          ))
        ]
      }
    })
```

Reply:
Yea, I kinda did this as an experiment so my researcher codesmell is on full display here lol. This is :fire:  and absolutely the right way to do it

Reply:
lol, must of the  _repo_ is "researcher codesmell" at this point in time :)